pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -709,3 +719,47 @@ func (iqr *IQR) AsResult() (*structs.PipeSearchResponseOuter, error) {
 
 	return response, nil
 }
+
+func (iqr *IQR) AddColumnWithDefaultValue(cname string, value utils.CValueEnclosure) error {
+	if err := iqr.validate(); err != nil {
+		log.Errorf("IQR.AddColumnWithDefaultValue: validation failed: %v", err)
+		return err
+	}
+
+	if iqr.mode == notSet {
+		return nil
+	}
+
+	if _, ok := iqr.deletedColumns[cname]; ok {
+		return fmt.Errorf("IQR.AddColumnWithDefaultValue: column %s is deleted", cname)
+	}
+
+	values, ok := iqr.knownValues[cname]
+	if !ok {
+		values = make([]utils.CValueEnclosure, iqr.NumberOfRecords())
+	}
+
+	for i := range values {
+		values[i] = value
+	}
+	iqr.knownValues[cname] = values
+
+	return nil
+}
+
+func (iqr *IQR) AddNewCreatedColumn(cname string) error {
+	if err := iqr.validate(); err != nil {
+		log.Errorf("IQR.AddNewCreatedColumn: validation failed: %v", err)
+		return err
+	}
+
+	if iqr.mode == notSet {
+		return nil
+	}
+
+	if _, ok := iqr.deletedColumns[cname]; ok {
+		return fmt.Errorf("IQR.AddNewCreatedColumn: column %s is deleted", cname)
+	}
+
+	return nil
+}
```

Comment:
I don't think this is used. If not, delete it

Reply:
Yes, deleted.